controllers/adminController.ts

```diff
@@ -136,3 +138,38 @@ export const saveUserSkills = async (req: Request, res: Response, next: NextFunc
     next(err);
   }
 };
+
+export const registerHr = async (req: Request, res: Response, next: NextFunction) => {
+  const { email, fullName, company, maxReservedStudents } = req.body;
+
+  const randomPassword = Math.random().toString(36).slice(-8);
+
+  try {
+    const newUser = new UserDb({
+      email,
+      password: await hashPwd(randomPassword),
+      role: 'HR',
+      fullName,
+      company,
+      maxReservedStudents,
+    });
+
+    const newHr = await UserDb.createNewUser(newUser, newUser.role);
+
+    const token = genToken(newHr._id, process.env.EXPIRES_REGISTER_TOKEN);
+    newHr.token = token;
+    await newHr.save();
+
+    const link = `${process.env.CORS_ORIGIN}/reset-password/${token}`;
+
+    await handleEmail({
+      to: newHr.email,
+      subject: 'Rejestracja użytkownika',
+      html: registerEmailTemplate(link),
+    });
+
```

Comment:
zastanawiam sie tu, jak usprawnić ten proces, aby admin za długo nie czekał na odpowiedź, po tym jak wszyscy uzytkownicy sie dodadza do bazy i otrzymaja maile-moglibyśmy wykorzystac EventEmitter, on moze przyspieszyc wysyłke maili i zapisywania użytkowników, poprzez umożliwienie równoczesnego wykonywania tych zadań w tle.

Reply:
Tu jest rejestracja Hr, oni są dodawani pojedynczo wiec tu nie będzie problemu ale w kodzie wyżej gdzie zapisujemy dane z pliku .csv i rozsyłamy linki może występować ten problem. Tylko nie używałem dawno EventEmitera ale z tego co rozumiem to on dział tak, że reaguje automatycznie na jakiś zdarzenie z innej części aplikacji. Nie widzę za bardzo tutaj zastosowania do tego. Jest Promise.allSettled() który możemy użyć zamiast Promise.all(). Umożliwiłoby to jednoczesne wykonywanie wszystkich obietnic, zamiast czekać na zakończenie jednej przed rozpoczęciem następnej.  To możemy przetestować ale nigdy tego nie używałem wiec może po testach zobaczymy jak faktycznie to działa i będziemy wdrażać jakieś rozwiązania.

Reply:
tak właśnie o plik csv mi chodziło :) z tego co czytałam mozna dołożyc jakby dodatkowy wątek,  emitujemy zdarzenie z danymi nowo zarejestrowanego użytkownika, a listener jak dostanie sygnał zajmuje się wysyłką wiadomosci, i mozna wiele czasu dzieki temu zaoszczedzic, bo ta druga funkcja działa w tle, Przemek mi to pokazywał w kodzie to jest ogromna różnica. Jasne na etap testów bedzie super to sprawdzic :)